models/Search/Backend/Data/Listing/Dao.php

```diff
@@ -36,16 +37,8 @@ public function load()
         $data = $this->db->fetchAll('SELECT * FROM search_backend_data' .  $this->getCondition() . $this->getGroupBy() . $this->getOrder() . $this->getOffsetLimit(), $this->model->getConditionVariables());
 
         foreach ($data as $entryData) {
-            if ($entryData['maintype'] == 'document') {
-                $element = Document::getById($entryData['id']);
-            } elseif ($entryData['maintype'] == 'asset') {
-                $element = Asset::getById($entryData['id']);
-            } elseif ($entryData['maintype'] == 'object') {
-                $element = DataObject::getById($entryData['id']);
-            } else {
-                $element = null;
-                Logger::err('unknown maintype ');
-            }
+            $element = Service::getElementById($entryData['maintype'], $entryData['id']);
```

Comment:
please add `$entryData['maintype']` check and log error if unknown

Reply:
I add a check